### PR TITLE
MM-11301: Validates listen address config value.

### PR DESCRIPTION
**Pull request by mkraft**
#### Summary
Validates listen address config value.

#### Ticket Link
[MM-11301](https://mattermost.atlassian.net/browse/MM-11301)

#### Checklist
- [x] Added or updated unit tests (required for all new features)

### CODE REVIEW

**Comment by mkraft** [comment]
@grundleborg @jespino Am I correct that `ListenAddress` would not use a domain but rather an IP and port?

File: model/config.go [hunk 1]
```diff
@@ -2326,7 +2326,8 @@ func (ss *ServiceSettings) isValid() *AppError {
 		}
 	}
 
-	if len(*ss.ListenAddress) == 0 {
+	listenAddressRE := regexp.MustCompile(`^(\d{1,3}.\d{1,3}.\d{1,3}.\d{1,3})?:\d{1,5}$`)
```
Comment (jespino) [inline model/config.go]: I would say it is enough, but you can still use 999.999.999.999 as IP which is clearly a not valid ip, or a :99999 which isn't a valid port. You can see more restrictive examples for IP address here: https://www.regular-expressions.info/ip.html

**Comment by mkraft** [comment]
@jespino I decided to add support for IPv6 addresses and drop regexes completely.

File: model/config.go [hunk 2]
```diff
@@ -2326,7 +2329,15 @@ func (ss *ServiceSettings) isValid() *AppError {
 		}
 	}
 
-	if len(*ss.ListenAddress) == 0 {
+	host, port, err := net.SplitHostPort(*ss.ListenAddress)
```
Comment (jespino) [inline model/config.go]: You are not handling the error here

**Comment by jespino** [comment]
I think I was wrong here about domains. Listen implementation receive a network type and an address, and the address is an IP or a Domain. so... we can have users using domains here. That means that we can receive there domain:port. So, we have to validate that host is a valid formatted address, or a valid formatted domain. I don't know which is the best approach here. maybe you can copy the golang isDomainName code in net package (here: https://golang.org/src/net/dnsclient.go#L119).

File: model/config.go (hunk 2 above)
Comment (mkraft) [reply model/config.go]: @jespino It is handling it down below once because they'll both return the same `AppError` so no need to do two returns.

**Pull request merged**